File: src/product/Item.java
package product;

public class Item {
    private String name;
    private int amount;

    public Item(String name, int amount) {
        this.name = name;
        this.amount = amount;
    }

    public int getAmount() {
        return amount;
    }

    public String getName() {
        return name;
    }

    public void setAmount(int amount) {
        this.amount = amount;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof Item)) {
            return false;
        }
        Item comparItem = (Item) obj;
        return this.name.equals(comparItem.getName());
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public String toString() {
        return "name: " + name + ", " + "amount: " + amount;
    }
}

File: TYPESCRIPT/src/product/Item.ts
export class Item {
  private _name: string;

  private _amount: number;

  constructor(name: string, amount: number) {
    this._name = name;
    this._amount = amount;
  }

  public get getAmount(): number {
    return this._amount;
  }

  public get getName(): string {
    return this._name;
  }

  public set setAmount(value: number) {
    this._amount = value;
  }

  public set setName(value: string) {
    this._name = value;
  }

  public equals(object: object): boolean {
    if (!(object instanceof Item)) {
      return false;
    }
    return this._name === object._name;
  }

  public toString(): string {
    return `name: ${this._name}, amount: ${this._amount}`;
  }
}
